### PR TITLE
Add serde modules for public/private/signaling keys

### PR DESCRIPTION
**Pull request by gferon**
This helps with serializing a struct containing all the registration data easily (like in `presage`).

### CODE REVIEW

File: libsignal-service/src/push_service.rs [hunk 1]
```diff
@@ -147,6 +148,42 @@ impl ProfileKey {
...
+                .try_into()
+                .map_err(|buf: Vec<u8>| {
+                    serde::de::Error::invalid_length(
+                        buf.len(),
+                        &"invalid profile key length",
+                    )
+                })?,
+        ))
+    }
+}
```
Comment (rubdos) [inline libsignal-service/src/push_service.rs:185]: I can't imagine that this isn't possible with the `with_base64` module that we already have and maybe a `#[serde(flatten)]`?

Comment (gferon) [reply libsignal-service/src/push_service.rs:185]: Our `serde_base64` will not deserialize to an array of known size, but to a `Vec<u8>` so even though `Serialize` can be derived, `Deserialize` can't.

Comment (rubdos) [reply libsignal-service/src/push_service.rs:185]: Ah right, and for a module to do so, you probably need HKT. Dismissed!

**Comment by rubdos** [comment]
LGTM

**Pull request merged**